03-GettingStarted/06-http-streaming/README.md

```diff
@@ -0,0 +1,427 @@
+|------------------------|-------------------------------|-------------------------------------|
+| Main response          | Chunked                       | Single, at end                      |
+| Progress updates       | Sent as data chunks           | Sent as notifications               |
+| Client requirements    | Must process stream           | Must implement message handler      |
+| Use case               | Large files, AI token streams | Progress, logs, real-time feedback  |
+
+### Key Differences Observed
+
+1. **Communication Pattern:**
+   - Classic HTTP streaming: Uses simple chunked transfer encoding to send data in chunks
+   - MCP streaming: Uses a structured notification system with JSON-RPC protocol
+
+2. **Message Format:**
+   - Classic HTTP: Plain text chunks with newlines
+   - MCP: Structured LoggingMessageNotification objects with metadata
+
+3. **Client Implementation:**
+   - Classic HTTP: Simple client that processes streaming responses
+   - MCP: More sophisticated client with a message handler to process different types of messages
+
+4. **Progress Updates:**
+   - Classic HTTP: The progress is part of the main response stream
+   - MCP: Progress is sent via separate notification messages while the main response comes at the end
+
+### Recommendations
+
+Based on what we've observed:
+
+1. **For simple streaming needs:** Classic HTTP streaming is simpler to implement and sufficient for basic streaming needs.
+
+2. **For complex, interactive applications:** MCP streaming provides a more structured approach with richer metadata and separation between notifications and final results.
+
+3. **For AI applications:** MCP's notification system is particularly useful for long-running AI tasks where you want to keep users informed of progress.
+
+---
+
+## 3. Streaming in MCP
+
+In MCP, streaming is not about sending the main response in chunks, but about sending **notifications** to the client while a tool is processing a request. These notifications can include progress updates, logs, or other events.
+
+#### Key differences from traditional streaming
+- The main result is still sent as a single response
+- Notifications are sent as separate messages during processing
+- The client must be able to handle and display these notifications
+
+#### What is a Notification?
+A notification is a message sent from the server to the client to inform about progress, status, or other events during a long-running operation. Notifications improve transparency and user experience.
+
+##### Why use notifications?
+- Keep users informed about progress
+- Enable real-time feedback and better UX
+- Useful for debugging and monitoring
+
+---
+
+## 4. Implementing Notifications in MCP
+
+### Server-side: Sending Notifications
+To send notifications from your MCP tool, use the context object (usually `ctx`) to call methods like `ctx.info()` or `ctx.log()`. These send messages to the client as the server processes a request.
```

Comment:
if you have the info, I would explain what the difference is between info and log and other types of messages. As a reader I'm looking cool, many ways to send a message, but which one to choose?